File: app/schemas/hr_process_audit.py
"""HR Process Audit schemas."""

from __future__ import annotations

from typing import Literal, Optional
from uuid import UUID

from pydantic import BaseModel, Field

AuditFlag = Literal[
    "incomplete_onboarding",
    "missing_exit",
    "orphan_placement",
    "exit_done_still_active",
]


class ProcessAuditItem(BaseModel):
    flag: AuditFlag
    severity: Literal["high", "medium", "low"] = "medium"
    title: str
    subject_name: str
    subject_user_id: Optional[UUID] = None
    checklist_id: Optional[UUID] = None
    exit_interview_id: Optional[UUID] = None
    detail: str
    deep_link: str
    anchor_date: Optional[str] = None


class ProcessAuditRead(BaseModel):
    as_of: str
    sla_days: int
    total: int
    counts: dict[str, int] = Field(default_factory=dict)
    items: list[ProcessAuditItem] = Field(default_factory=list)

File: frontend/src/api/hrProcessAudit.ts
import { apiClient } from './client';

export type ProcessAuditFlag =
  | 'incomplete_onboarding'
  | 'missing_exit'
  | 'orphan_placement'
  | 'exit_done_still_active';

export interface ProcessAuditItem {
  flag: ProcessAuditFlag;
  severity: 'high' | 'medium' | 'low';
  title: string;
  subject_name: string;
  subject_user_id: string | null;
  checklist_id: string | null;
  exit_interview_id: string | null;
  detail: string;
  deep_link: string;
  anchor_date: string | null;
}

export interface ProcessAudit {
  as_of: string;
  sla_days: number;
  total: number;
  counts: Record<string, number>;
  items: ProcessAuditItem[];
}

export const hrProcessAuditApi = {
  get: async (slaDays = 14): Promise<ProcessAudit> => {
    const { data } = await apiClient.get<ProcessAudit>('/hr/process-audit', {
      params: { sla_days: slaDays },
    });
    return data;
  },
};
